framework/data_types/ndarray.h

```diff
@@ -19,23 +22,22 @@ class NDArray
 {
 private:
   size_t rank_;
-  std::vector<size_t> dimensions_;
-  std::vector<size_t> strides_;
   size_t size_;
-  T* base_;
-
-  template <bool...>
-  struct bool_pack
-  {
-  };
-
-  template <class... U>
-  using conjunction = std::is_same<bool_pack<true, U::value...>, bool_pack<U::value..., true>>;
-
-  template <typename... U>
-  using AllIntegral = typename conjunction<std::is_integral<U>...>::type;
+  std::unique_ptr<T[]> storage_;
+  static constexpr size_t max_rank = 10;
```

Comment:
I just looked back at the coding standards and I could have sworn we had added something about class declaration structure along the lines to have the structure of routines -> attributes -> static routines -> static attributes with each having public/protected/private. I was going to ask if the declaration could be organized in this way, but I don't see anything on this in the coding standards... Does anyone else remember this discussion from long ago?

Reply:
I remember the discussion and I do not see it in the coding standards either. Also, no issue capturing this either...

Reply:
I've made changes to restructure the code following this coding standard.